docs/extending/customizing_calendar.md

```diff
@@ -0,0 +1,282 @@
+# Customizing Calendar widget
+
+By default, the Calendar widget enables you to display all scheduled events and perform basic actions on them.
+You can also configure it to display your custom event types or display them from [custom sources](#configuring-event-sources).
+
+Optionally, you can [change the colors and icons](#customizing-colors-and-icons) and make the widget look differently depending on the [SiteAccess configuration](../siteaccess.md#configuring-siteaccesses).
+
+## Configuring custom events
+
+This example shows how to create custom Events which add Bank holidays to the calendar.
+
+First, create a new event in `src/Calendar/Holidays/Event.php`:
+
+``` php
+<?php
+
+namespace App\Calendar\Holidays;
+
+use DateTimeInterface;
+use EzSystems\EzPlatformCalendar\Calendar\Event as CalendarEvent;
+
+class Event extends CalendarEvent
+{
+    public function __construct(string $id, DateTimeInterface $dateTime, EventType $type)
+    {
+        parent::__construct($type, $id, $dateTime);
+    }
+}
+```
+
+Here, you define a new class for your event based on `EzSystems\EzPlatformCalendar\Calendar\Event`.
+
+Next, create `src/Calendar/Holidays/EventType.php`:
+
+```php hl_lines="23 24 25 26"
+<?php
+
+namespace App\Calendar\Holidays;
+
+use EzSystems\EzPlatformCalendar\Calendar\Event;
+use EzSystems\EzPlatformCalendar\Calendar\EventAction\EventActionCollection;
+use EzSystems\EzPlatformCalendar\Calendar\EventType\EventTypeInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class EventType implements EventTypeInterface
+{
+    private const EVENT_TYPE_IDENTIFIER = 'holiday';
+
+    private $translator;
+    private $actions;
+
+    public function __construct(iterable $actions, TranslatorInterface $translator)
+    {
+        $this->actions = new EventActionCollection($actions);
+        $this->translator = $translator;
+    }
+
+    public function getTypeIdentifier(): string
+    {
+        return self::EVENT_TYPE_IDENTIFIER;
+    }
+
+    public function getTypeLabel(): string
+    {
+        return $this->translator->trans(
+            /** @Desc("Bank holidays") */
+            'bank_holiday.label',
+            [],
+            'app_calendar_events'
+        );
+    }
+
+    public function getEventName(Event $event): string
+    {
+        return $event->getId();
+    }
+
+    public function getActions(): EventActionCollection
+    {
+        return $this->actions;
+    }
+}
+```
+
+The identifier defined in lines 23-26 can be used later in [configuration](#customizing-colors-and-icons).
+
+Complete the procedure by registering the new event in `config/services.yaml`:
+
+``` yaml
+services:
+    // ...
+    App\Calendar\Holidays\EventType:
+        arguments:
+            $actions: []
+        tags:
+            - { name: ezplatform.calendar.event_type }
+```
+
+## Configuring event sources
+
+To add specific Events to your calendar, you need to create a source factory.
```

Comment:
```suggestion
To add specific Events to your calendar, you need to create a event source.
```
Just to clarify: `EventSourceFactory` is not the part of Calendar API. It's role is to help create `InMemoryEventSource` in order to be able to register it as service in DIC.  

What you actually need to do is provide an `\EzSystems\EzPlatformCalendar\Calendar\EventSource\EventSourceInterface` implementation  e.g `InMemoryEventSource`  However In the real life scenarios usage of `InMemorySourceProvider`  is limited so we should provide any example how to implement `EventSourceInterface` from scratch. 

Also worth to mention is `\EzSystems\EzPlatformCalendar\Calendar\LanguageBasedEvent` interface. In pratice it allows to filter events by the language (`InMemoryEventSource` will take it into the account).

Reply:
Ah, interesting... could you elaborate on why you consider `InMemoryEventSource` as limited in real life ?

Reply:
The main limitation is memory usage. `InMemoryEventSource` stores all available entries in- memory and filter out the events which don't match the requested criteria in runtime (e.g. time range, type, pagination limit). It's optimal for small datasets e.g. bank holidays, employee birthdays and is super easy to implement. 

In the case of 
* datasets that grow over time 
* " heavy" event objects (in terms of memory or instantiation time)
* 3-rd party API limits 
you won't be able to load all data into memory at once or this will be not optimal e.g. content scheduled for publication/hide in a large organization. 

This problem solves custom `EventSourceInterface` implementation: you are loading only events that are necessary (e.g. only from the specific time frame) however it requires much more effort from the developer.